client/handle_http.go

```diff
@@ -926,6 +926,10 @@ func (te *TransferEngine) runMux() error {
 			// If no transfers were created and we have an error, the job is no
 			// longer active
 			if job.job.lookupErr != nil && job.job.totalXfer == 0 {
+				// Make sure we update results of this error
+				log.Errorln(job.job.lookupErr)
+				results := &TransferResults{jobId: job.job.uuid, job: job.job, Error: job.job.lookupErr, Scheme: job.job.remoteURL.Scheme}
+				te.resultsMap[job.uuid] <- results
```

Comment:
We can't do blocking operations in the runMux loop.  That will lead to deadlocks, no?

I'd need to page the specific knowledge back in but I recall some issues in having non-transfer results sent in the transfer result channel.

Can the plugin check for the error condition when the channel closes perhaps?

Reply:
Do you have any suggestions for how to have the plugin check for this error condition? Just not sure how I can check for error conditions if the channel is already closed and the context has no errors attached to it. Should I maybe have something for the client and check for a tc.Error or something?